tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -64,6 +65,10 @@ private Tracer() {}
     // when our observers are modified.
     private static volatile Consumer<Span> compositeObserver = _span -> {};
 
+    private static final Map<String, TraceLocalObserver<?>> traceLocalObservers = new HashMap<>();
+    // see above comment on compositeObserver
+    private static volatile Consumer<TraceState> compositeTraceLocalObserver = _traceState -> {};
```

Comment:
tracelocal listeners should be fully encapsulated within the TraceLocal object itself (An overridable method, or custom listener interface would handle this nicely)

Reply:
yeah, fair enough - done